### PR TITLE
Sl fix online upgrade

### PR DESCRIPTION
**Pull request by shuolinl**

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/executor/ConfigPlanExecutor.java [hunk 1]
```diff
@@ -362,6 +362,17 @@ public TSStatus executeNonQueryPlan(ConfigPhysicalPlan physicalPlan)
...
+      case CreateRoleDep:
+      case DropUserDep:
+      case DropRoleDep:
+      case GrantRoleDep:
+      case GrantUserDep:
+      case GrantRoleToUserDep:
+      case RevokeUserDep:
+      case RevokeRoleDep:
+      case RevokeRoleFromUserDep:
+      case UpdateUserDep:
```
Comment (shuolinl) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/executor/ConfigPlanExecutor.java:375]: 这里需要再补全一下。

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/IOUtils.java [hunk 2]
```diff
@@ -128,13 +154,26 @@ public static String readString(
    * @throws IOException when an exception raised during operating the stream.
    */
   public static PathPrivilege readPathPrivilege(
-      DataInputStream inputStream, String encoding, ThreadLocal<byte[]> strBufferLocal)
+      DataInputStream inputStream,
+      String encoding,
+      ThreadLocal<byte[]> strBufferLocal,
+      boolean compatible)
```
Comment (JackieTien97) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/IOUtils.java:160]: add java doc for this parameter

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/IOUtils.java [hunk 3]
```diff
@@ -117,6 +118,31 @@ public static String readString(
     return null;
   }
 
+  public static Pair<String, Boolean> readAuthString(
```
Comment (JackieTien97) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/IOUtils.java:121]: add java doc for this method,  at least that like: for IOTDB 1.2, the username's length will be stored as a negative number.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/executor/ConfigPlanExecutor.java [hunk 4]
```diff
@@ -362,6 +362,17 @@ public TSStatus executeNonQueryPlan(ConfigPhysicalPlan physicalPlan)
       case RevokeRole:
       case RevokeRoleFromUser:
       case UpdateUser:
+      case CreateUserDep:
```
Comment (JackieTien97) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/executor/ConfigPlanExecutor.java:365]:
```suggestion
      // Dep for Deprecated
      case CreateUserDep:
```

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java [hunk 5]
```diff
@@ -183,7 +183,21 @@ public TSStatus authorNonQuery(AuthorPlan authorPlan) {
...
+        case DropRoleDep:
+        case GrantUserDep:
+        case GrantRoleDep:
+        case GrantRoleToUserDep:
+        case RevokeUserDep:
+        case RevokeRoleDep:
+        case RevokeRoleFromUserDep:
+          throw new AuthException(
+              TSStatusCode.UNSUPPORTED_OPERATION,
+              "unsupport operation: " + authorPlan.getAuthorType());
```
Comment (JackieTien97) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java:198]:
```suggestion
          // TODO will deal with these authority upgrade soon in the future.
          throw new AuthException(
              TSStatusCode.UNSUPPORTED_OPERATION,
              "unsupport operation: " + authorPlan.getAuthorType());
```

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

**Comment by shuolinl** [comment]
刚刚 force push 丢了一个 commit， 在另一台机器上，一会把他 cherry-pick 过来

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**